environs/simplestreams/datasource.go

```diff
@@ -73,26 +76,55 @@ type urlDataSource struct {
 	requireSigned        bool
 }
 
-// NewURLDataSource returns a new datasource reading from the specified baseURL.
-func NewURLDataSource(description, baseURL string, hostnameVerification utils.SSLHostnameVerification, priority int, requireSigned bool) DataSource {
-	return &urlDataSource{
-		description:          description,
-		baseURL:              baseURL,
-		hostnameVerification: hostnameVerification,
-		priority:             priority,
-		requireSigned:        requireSigned,
+// Config has values to be used in constructing a datasource.
+type Config struct {
+	// Description of the datasource
+	Description string
+
+	// BaseURL is the URL for this datasource.
+	BaseURL string
+
+	// HostnameVerification indicates whether to use self-signed credentials
+	// and not try to verify the hostname on the TLS/SSL certificates.
+	HostnameVerification utils.SSLHostnameVerification
+
+	// PublicSigningKey is the public key used to validate signed metadata.
+	PublicSigningKey string
+
+	// Priority is an importance factor for the datasource. Higher number means
+	// higher priority. This is will facilitate sorting data sources in order of
+	// importance.
+	Priority int
+
+	// RequireSigned indicates whether this datasource requires signed data.
+	RequireSigned bool
+}
+
+// Validate() checks that the baseURL is valid and the description is set.
```

Comment:
Small nitpick: parens can be omitted here if we want to be consistent

Reply:
done